### PR TITLE
Cypress/fe-1592

### PR DESCRIPTION
**Pull request by marcinstachowicz**
# Cypress BDD test scenarios
## New Scenarios
for Number Input, Pager, Pages

### smaller improvements
labelWidth, labelAlign

### CODE REVIEW

File: cypress/integration/pager.feature [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+      | totalRecords |
+      | -1           |
+      | -10          |
+      | -100         |
+
+  @negative
+  Scenario Outline: Set totalRecords out of scope
+    When I set totalRecords to "<totalRecords>"
+    Then totalRecords is set to ""
+      And I am on 1st of "1" pages
```
Comment (DlgSHi) [inline cypress/integration/pager.feature]: why we should pass parameter "1" here ?

File: cypress/integration/pager.feature [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    Examples:
+      | Sample text             |
+      | áéíóú¿¡üñ               |
+      | !@#$%^*()_+-=~[];:.,?{} |
+      | ÄÖÜßäöüß                |
+      | <>                      |
+
+  @positive
+  Scenario Outline: Set pageSize to <pageSize>
```
Comment (DlgSHi) [inline cypress/integration/pager.feature]: page size -> separately

File: cypress/integration/pager.feature [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    When I check showPageSizeSelection checkbox
+    And I uncheck showPageSizeSelection checkbox
+    Then pageSize is not visible
+
+  @positive
+  Scenario: Pagination previous arrow is disabled
+    #  When I open "Pager" component page
+    Then previous pagination arrow is disabled
+
+    @positive
```
Comment (DlgSHi) [inline cypress/integration/pager.feature]: formatting

File: cypress/integration/pager.feature [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+  Scenario: Pagination next arrow is disabled
+    When I paginate next 9 times
+    Then next pagination arrow is disabled
+
+      @positive
```
Comment (DlgSHi) [inline cypress/integration/pager.feature]: formatting

File: cypress/integration/pager.feature [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+  Scenario: Pagination previous arrow is disabled after previous paginate
+    When I type "10" to input pagination
+    And I paginate previous 9 times
```
Comment (DlgSHi) [inline cypress/integration/pager.feature]: paginate previous or click previous arrow?

File: cypress/support/step-definitions/pages-steps.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+Then('My {word} Page is not visible', () => {
+  titleNoIframe().should('not.exist');
+});
+
+When('I go to {word} page', (word) => {
+  dataComponentButtonByText(`Go to ${word} page.`).click();
+});
+
+When('I open no iFrame component preview', () => {
+  dataComponentButtonByText('Open Preview').click();
```
Comment (DlgSHi) [inline cypress/support/step-definitions/pages-steps.js]: why U didn't add normal locator for an Open Preview?

File: cypress/support/step-definitions/pages-steps.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+});
+
+When('I close page', () => {
+  cy.get('[data-element="close"]').click();
```
Comment (DlgSHi) [inline cypress/support/step-definitions/pages-steps.js]: why U didn't add normal locator for an Open Preview?

File: cypress/integration/numberInput.feature [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+      | áéíóú¿¡üñ               |
+      | !@#$%^*()_+-=~[];:.,?{} |
+      | ÄÖÜßäöüß                |
+      | <>                      |
+
+  @ignore
+  # currently out of scope
+  Scenario: Time to disappear
+    When I set timeToDisappear to "<timeToDisappear>"
+    Then timeToDisappear is set
```
[the remark this answers is not in the document]
Comment (marcinstachowicz) [reply cypress/integration/numberInput.feature]: fixed

File: cypress/integration/buttonToggleGroup.feature [hunk 9]
```diff
@@ -78,7 +78,7 @@ Feature: Button Toggle Group component
   @positive
   Scenario Outline: Change Button Toggle Group label width
     When I check labelInline checkbox
-      And I set label width to "<width>"
+      And I set labelWidth to "<width>"
     Then Label width is set to "<width>"
```
[the remark this answers is not in the document]
Comment (marcinstachowicz) [reply cypress/integration/buttonToggleGroup.feature]: fixed

File: cypress/integration/numberInput.feature [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+      | 1234567890              |
+      | áéíóú¿¡üñ               |
+      | !@#$%^*()_+-=~[];:.,?{} |
+      | ÄÖÜßäöüß                |
+      | <>                      |
+
+  @positive
+  Scenario: Enable label inline
+    When I check labelInline checkbox
+    Then GroupedCharacter labelInline is enabled
```
[the remark this answers is not in the document]
Comment (marcinstachowicz) [reply cypress/integration/numberInput.feature]: fixed

File: cypress/integration/numberInput.feature [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+
+  @positive
+  Scenario: Disable label inline
+    When I check labelInline checkbox
+      And I uncheck labelInline checkbox
+    Then GroupedCharacter labelInline is disabled
```
[the remark this answers is not in the document]
Comment (marcinstachowicz) [reply cypress/integration/numberInput.feature]: fixed

File: cypress/integration/numberInput.feature [hunk 12]
```diff
@@ -0,0 +1,119 @@
...
+
+  @positive
+  Scenario Outline: Set label width
+    When I check labelInline checkbox
+      And I set labelWidth to "<labelWidth>"
+    Then GroupedCharacter labelWidth is set to "<labelWidth>"
```
[the remark this answers is not in the document]
Comment (marcinstachowicz) [reply cypress/integration/numberInput.feature]: fixed

File: cypress/integration/pager.feature (hunk 3 above)
Comment (marcinstachowicz) [reply cypress/integration/pager.feature]: fixed

File: cypress/integration/pager.feature (hunk 4 above)
Comment (marcinstachowicz) [reply cypress/integration/pager.feature]: fixed

File: cypress/support/step-definitions/pages-steps.js (hunk 6 above)
Comment (marcinstachowicz) [reply cypress/support/step-definitions/pages-steps.js]: added const text called 'Open Preview'

File: cypress/support/step-definitions/pages-steps.js (hunk 7 above)
Comment (marcinstachowicz) [reply cypress/support/step-definitions/pages-steps.js]: fixed

File: cypress/integration/pager.feature (hunk 5 above)
Comment (marcinstachowicz) [reply cypress/integration/pager.feature]: paginate is ok

File: cypress/integration/pager.feature [hunk 13]
```diff
@@ -0,0 +1,92 @@
+Feature: Pager component
+  I want to change Pager component properties
+
+  Background: Open Pager component page
+    Given I open "Pager" component page
+
+    @positive
+    Scenario Outline: Set totalRecords to <totalRecords>
```
[the remark this answers is not in the document]
Comment (marcinstachowicz) [reply cypress/integration/pager.feature]: we talked about it

File: cypress/integration/pager.feature (hunk 1 above)
Comment (marcinstachowicz) [reply cypress/integration/pager.feature]: resolved

**Pull request merged**